src/panels/lovelace/hui-header.ts

```diff
@@ -0,0 +1,750 @@
+          </ha-list-item>`
+        );
+      });
+      result.push(
+        html`<ha-button-menu slot="actionItems">
+          <ha-icon-button
+            slot="trigger"
+            .label=${this.hass!.localize("ui.panel.lovelace.editor.menu.open")}
+            .path=${mdiDotsVertical}
+          ></ha-icon-button>
+          ${listItems}
+        </ha-button-menu>`
+      );
+    }
+    return html`${result}`;
+  }
+
+  protected render(): TemplateResult {
+    const views = this.lovelace?.config.views ?? [];
+    const curViewConfig =
+      typeof this.curView === "number" ? views[this.curView] : undefined;
+    const dashboardTitle = this.panel
+      ? getPanelTitle(this.hass, this.panel)
+      : undefined;
+
+    const _isTabHiddenForUser = (view: LovelaceViewConfig) =>
+      view.visible !== undefined &&
+      ((Array.isArray(view.visible) &&
+        !view.visible.some((e) => e.user === this.hass!.user?.id)) ||
+        view.visible === false);
+
+    const visibleViews = views.filter(
+      (view) => !view.subview && !_isTabHiddenForUser(view)
+    );
+
+    return html`
+      <div
+        class=${classMap({
+          "edit-mode": this._editMode,
+        })}
+      >
+        <div class="header">
+          <div class="toolbar">
+            ${this._editMode
+              ? html`
+                  <div class="main-area">
+                    <div class="main-title">
+                      ${dashboardTitle ||
+                      this.hass!.localize("ui.panel.lovelace.editor.header")}
+                      <ha-icon-button
+                        .label=${this.hass!.localize(
+                          "ui.panel.lovelace.editor.edit_lovelace.edit_title"
+                        )}
+                        .path=${mdiPencil}
+                        class="edit-icon"
+                        @click=${this._editDashboard}
+                      ></ha-icon-button>
+                    </div>
+                  </div>
+                  <div class="action-items">${this._renderActionItems()}</div>
+                `
+              : html`
+                  ${curViewConfig?.subview
+                    ? html`
+                        <ha-icon-button-arrow-prev
+                          slot="navigationIcon"
+                          @click=${this._goBack}
+                        ></ha-icon-button-arrow-prev>
+                      `
+                    : html`
+                        <ha-menu-button
+                          slot="navigationIcon"
+                          .hass=${this.hass}
+                          .narrow=${this.narrow}
+                        ></ha-menu-button>
+                      `}
+                  <div class="main-area">
+                    ${curViewConfig?.subview
+                      ? html`<div class="main-title">
+                          ${curViewConfig.title}
+                        </div>`
+                      : (!this.narrow || window.innerWidth > 600) &&
```

Comment:
Why do you use `window.innerWidth` here? For displaying the bottom bar you only use `this.narrow`.